tests/callbacks/test_callbacks.py

```diff
@@ -86,15 +86,15 @@ def test_trainer_callback_system(torch_save):
         call.on_before_zero_grad(trainer, model, trainer.optimizers[0]),
         call.on_batch_end(trainer, model),
         call.on_train_batch_end(trainer, model, ANY, ANY, 2, 0),
+        call.on_epoch_end(trainer, model),
+        call.on_train_epoch_end(trainer, model, ANY),
```

Comment:
I think that @williamFalcon had a point some time ago about training shall be till validation, and the example was with validation multiple times over long training...
cc: @tchaton @PyTorchLightning/core-contributors

Reply:
yes it still works like that only if `val_check_interval < 1.0` or it an int where `val_check_interval%num_training_batches != 0`. But if it is set to `1.0` then validation here happens after training_epoch because we create checkpoints in `on_validation_end` and epoch level learning rates are updated once training is done since in case of `ReduceLROnPlateau` we need to have the monitor metrics and they are only available after complete training is done in case monitor is training specific.